src/shipit_static_analysis/shipit_static_analysis/__init__.py

```diff
@@ -61,7 +61,7 @@ def build_lines_hash(self):
 
         # Only read necessary lines
         with open(full_path) as source:
-            start = self.line - 1
+            start = self.line > 0 and self.line - 1 or 0
```

Comment:
Nit: Another variant (which I find more readable, but not a strong opinion) could be `max(self.line - 1, 0)`.